connection.go

```diff
@@ -581,43 +620,83 @@ func pack(h *smallWBuf, enc *encoder, reqid uint32,
 	return
 }
 
-func (conn *Connection) writeAuthRequest(w *bufio.Writer, scramble []byte) (err error) {
+func (conn *Connection) writeRequest(w *bufio.Writer, req Request) (err error) {
 	var packet smallWBuf
-	req := newAuthRequest(conn.opts.User, string(scramble))
 	err = pack(&packet, newEncoder(&packet), 0, req, ignoreStreamId, conn.Schema)
 
 	if err != nil {
-		return errors.New("auth: pack error " + err.Error())
+		return fmt.Errorf("pack error %w", err)
```

Comment:
To avoid surprises, let's do it right as in the examples:
https://go.dev/blog/go1.13-errors

```suggestion
		return fmt.Errorf("pack error: %w", err)
```

and above.

In practice there is no difference:
https://play.golang.com/p/CHwfiMpL5DH
but it is more obvious from a client side.

Reply:
Changed